.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Dynatrace LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build and test
+on:
+  pull_request:
+    branches: [ "*" ]
```

Comment:
When you want to run a given workflow on all branches, you don't need to specify them. Following will work:
```
on:
  pull_request:
  push:
  workflow_dispatch:
  workflow_call:
```

Reply:
In such scenario, push will work also on tags